api/valor_api/backend/metrics/detection.py

```diff
@@ -645,6 +655,615 @@ def _compute_detailed_curves(
+    table,
+):
+    """Get the appropriate column for calculating IOUs from models.Annotation."""
+    match annotation_type:
+        case AnnotationType.BOX:
+            box = table.box
+        case AnnotationType.POLYGON:
+            box = gfunc.ST_Envelope(table.polygon)
+        case AnnotationType.RASTER:
+            box = gfunc.ST_Envelope(gfunc.ST_MinConvexHull(table.raster))
+        case _:
+            raise RuntimeError
+    return gfunc.ST_AsGeoJSON(box)
+
+
+def wkt_to_array(wkt_data) -> np.ndarray:
+    """Convert a WKT string to an array of coordinates."""
+    coordinates = json.loads(wkt_data)["coordinates"][0]
+    return np.array(coordinates)
+
+
+def _calculate_bbox_intersection(bbox1, bbox2) -> float:
+    """Calculate the intersection between two bounding boxes."""
+
+    # Calculate intersection coordinates
+    xmin_inter = max(bbox1[:, 0].min(), bbox2[:, 0].min())
+    ymin_inter = max(bbox1[:, 1].min(), bbox2[:, 1].min())
+    xmax_inter = min(bbox1[:, 0].max(), bbox2[:, 0].max())
+    ymax_inter = min(bbox1[:, 1].max(), bbox2[:, 1].max())
+
+    # Calculate width and height of intersection area
+    width = max(0, xmax_inter - xmin_inter)
+    height = max(0, ymax_inter - ymin_inter)
+
+    # Calculate intersection area
+    intersection_area = width * height
+    return intersection_area
+
+
+def _calculate_bbox_union(bbox1, bbox2) -> float:
+    """Calculate the union area between two bounding boxes."""
+    area1 = (bbox1[:, 0].max() - bbox1[:, 0].min()) * (
+        bbox1[:, 1].max() - bbox1[:, 1].min()
+    )
+    area2 = (bbox2[:, 0].max() - bbox2[:, 0].min()) * (
+        bbox2[:, 1].max() - bbox2[:, 1].min()
+    )
+    union_area = area1 + area2 - _calculate_bbox_intersection(bbox1, bbox2)
+    return union_area
+
+
+def _calculate_bbox_iou(bbox1, bbox2) -> float:
+    """Calculate the IOU between two bounding boxes."""
+    intersection = _calculate_bbox_intersection(bbox1, bbox2)
+    union = _calculate_bbox_union(bbox1, bbox2)
+    iou = intersection / union
+    return iou
+
+
+def _get_joint_df(
+    db: Session,
+    groundtruth_filter: schemas.Filter,
+    prediction_filter: schemas.Filter,
+    grouper_mappings: dict,
+    annotation_type: AnnotationType,
+) -> pandas.DataFrame:
+    """Create a joint dataframe of groundtruths and predictions"""
+
+    gt = generate_select(
+        models.Dataset.name.label("dataset_name"),
+        models.GroundTruth.id.label("id"),
+        models.GroundTruth.annotation_id.label("annotation_id"),
+        models.Annotation.datum_id.label("datum_id"),
+        models.Datum.uid.label("datum_uid"),
+        case(
+            grouper_mappings["label_id_to_grouper_id_mapping"],
+            value=models.GroundTruth.label_id,
+        ).label("label_id_grouper"),
+        _annotation_type_to_geojson(annotation_type, models.Annotation).label(
+            "geojson"
+        ),
+        gfunc.ST_AsPNG(models.Annotation.raster).label("raster"),
```

Comment:
What happens if the annotation is a polygon or bounding box? Its not guaranteed (or that common) to have object detections defined by rasters.

Reply:
then this column would be None, right? there are other tests that check for non-raster cases